src/client.rs

```diff
@@ -153,33 +165,49 @@ impl ClientFlow {
         Ok(event)
     }
 
-    fn maybe_abort_command(&mut self, status: &Status) {
-        let Some((command_tag, _)) = self.send_command_state.command_in_progress() else {
-            return;
-        };
+    fn maybe_abort_command(
+        &mut self,
+        status: &Status,
+    ) -> Option<(Tag<'static>, ClientFlowCommandHandle)> {
+        let (command_tag, _) = self.send_command_state.command_in_progress()?;
 
-        // TODO(#45): Should we handle Status::No here?
-        // TODO(#46): Emit event with status and handle to client. Reuse StatusReceived? Introduce CommandRejected?
         match status {
             Status::Bad {
                 tag: Some(status_tag),
                 ..
-            } if status_tag == command_tag => {
-                self.send_command_state.abort_command();
-            }
-            _ => (),
+            } if status_tag == command_tag => self.send_command_state.abort_command(),
+            _ => None,
         }
     }
 }
 
+/// A handle for an enqueued [`Command`].
+///
+/// This handle can be used to track the sending progress. After a [`Command`] was enqueued via
+/// [`ClientFlow::enqueue_command`] it is in the process of being sent until
+/// [`ClientFlow::progress`] returns a [`ClientFlowEvent::CommandSent`] or
+/// [`ClientFlowEvent::CommandRejected`] with the corresponding handle.
 #[derive(Clone, Copy, Debug, Eq, PartialEq, Hash)]
 pub struct ClientFlowCommandHandle(u64);
 
 #[derive(Debug)]
 pub enum ClientFlowEvent {
+    /// The enqueued [`Command`] was sent successfully.
     CommandSent {
+        /// The handle of the enqueued [`Command`].
+        handle: ClientFlowCommandHandle,
+        /// The [`Tag`] of the enqueued [`Command`].
         tag: Tag<'static>,
+    },
+    /// The enqueued [`Command`] wasn't sent completely because the server rejected it.
+    CommandRejected {
+        /// The handle of the enqueued [`Command`].
         handle: ClientFlowCommandHandle,
+        /// The [`Tag`] of the enqueued [`Command`].
+        tag: Tag<'static>,
+        /// The [`Status`] sent by the server in order to reject the [`Command`].
+        /// This [`Status`] was already handled by [`ClientFlow`] and can safely be ignored.
```

Comment:
It could still have an `Code::Alert` telling the client to display an error message. Maybe we should say that the "flow", e.g., aborting the command was handled but the code and text could still be used.

Reply:
I didn't write that it's forbidden to use it. But still, I'll try to find a better phrase.

Reply:
Thanks :-)